docs/tutorials/elasticdl_deepctr.md

```diff
@@ -0,0 +1,279 @@
+Kaggle Advertising Challenge contest. There are 13 columns of integer
+features I1-I13 and 26 columns of categorical features C1-C26 in the dataset.
+
+## 1. Prerequisites
+
+- Install Minikube, preferably >= v1.11.0, following the installation guide.
+Minikube runs a single-node Kubernetes cluster in a virtual machine on your
+personal computer.
+- Install Docker CE, preferably >= 18.x, following the guide for building
+Docker images containing user-defined models and the ElasticDL framework.
+- Install Python, preferably >= 3.6, because the ElasticDL command-line
+tool is in Python.
+- Install [pyrecordio](https://pypi.org/project/pyrecordio/),
+[deepctr](https://pypi.org/project/deepctr/) and TensorFlow >= 2.0,
+[elasticdl_client](https://pypi.org/project/elasticdl-client/).
+
+## 2. Convert Data into RecordIO
+
+ElasticDL requires that the sample data can be queried by an integer
+index, so users need to convert the training data in [criteo_sample.txt](https://raw.githubusercontent.com/shenweichen/DeepCTR/master/examples/criteo_sample.txt)
+into RecordIO files.
+
+```python
+import argparse
+import os
+import pathlib
+import sys
+
+import recordio
+import tensorflow as tf
+
+COLUMNS = (
+    ["label"]
+    + ["I" + str(i) for i in range(1, 14)]
+    + ["C" + str(i) for i in range(1, 27)]
+)
+
+
+def convert_data_to_tf_example(sample_data, columns):
+    features = {}
+    column_data = sample_data.split(",")
+    for i, column_name in enumerate(columns):
+        value = column_data[i].strip()
+        if column_name[0] == "I" or column_name == "label":
+            value = 0.0 if value == "" else float(value)
+            feature = tf.train.Feature(
+                float_list=tf.train.FloatList(value=[value])
+            )
+        else:
+            feature = tf.train.Feature(
+                bytes_list=tf.train.BytesList(value=[value.encode("utf-8")])
+            )
+        features[column_name] = feature
+
+    example = tf.train.Example(
+        features=tf.train.Features(feature=features)
+    ).SerializeToString()
+
+    return example
+
+
+def convert_to_recordio_files(file_path, dir_name, records_per_shard=10240):
+    pathlib.Path(dir_name).mkdir(parents=True, exist_ok=True)
+
+    writer = None
+    with open(file_path, "r") as f:
+        f.readline() # Skip the header in the first line.
+        for index, row in enumerate(f):
+            if index % records_per_shard == 0:
+                if writer:
+                    writer.close()
+
+                shard = index // records_per_shard
+                file_path_name = os.path.join(dir_name, "data-%05d" % shard)
+                writer = recordio.Writer(file_path_name)
+            example = convert_data_to_tf_example(row, COLUMNS)
+            writer.write(example)
+
+        if writer:
+            writer.close()
+
+# We use the same dataset for training and valuation for simplity.
+convert_to_recordio_files("./criteo_sample.txt", "./data/criteo_records/train")
```

Comment:
We convert the same file into train&test dataset?

Reply:
Yes.